src/scripts/main.js

```diff
@@ -1,3 +1,27 @@
 'use strict';
 
-// write code here
+document.querySelectorAll('.tree li').forEach((li) => {
+  const childNodes = Array.from(li.childNodes);
+
+  childNodes.forEach((node) => {
+    if (node.nodeType === Node.TEXT_NODE && node.textContent.trim() !== '') {
```

Comment:
The condition checks for text nodes that are not empty. This is correct, but be aware that this will wrap every non-empty text node in a span, which might not be necessary for all cases.